### DISCUSSION TITLE
help update server context

### DISCUSSION THREAD

**Discussion by Radu-0** [pymodbus-dev/pymodbus#1175] [Q&A] [answered]
Hi,

I am trying to update the context of a Modbus RTU Serial Server.  I am using Python 3.11 and pymodbus 3.0.0 on Windows 10.

I have started from the latest example **server_async**. Then I implement update_writer similar as in older update_server example.

Unfortunately, it seems that I cannot run the update_writer and serialServer in the same time.., I have tried various ways: 

- Thread, 
- LoopingCall from twisted, 
- simply await for update_writer (the asyncio way)  

Each way I tried, either the server runs either the updater.. I don't see any alternating between the 2 calls... 
There are no errors or warning.. it looks like the 2 calls are waiting one for another to finish

I know there are major changes in pymodbus 3.0.0 and not sure if this is relevant.. but with Python 3.10 + pymodbus 2.5.3  + LoopingCall it worked but I am trying to move away from that config for other reasons...

Do you have any suggestions? 

Thank you!

**Comment by janiversen** [comment]
You need to run the update_writer as async so that it allows other tasks to work. I hope to get around to updating the example soon.

**Reply by Radu-0** [reply]
I did exactly that... 
 the 2 calls were done like:
 
     await updating_writer(context)
    server = await StartAsyncSerialServer(......
    
 no luck :-(

**Reply by janiversen** [reply]
The first await blocks until updating_writer returns, that is not what you want, updating_writer should run as a async_task.

**Comment by janiversen** [comment, marked as answer]
#1176 might be of help.

**Comment by Radu-0** [comment]
Indeed 3.0.2  + using the Task did the trick! Thank you!
